test/--version.bats

```diff
@@ -3,7 +3,7 @@
 load test_helper
 
 _setup() {
-  export GIT_DIR="${PYENV_TEST_DIR}/.git"
+  export GIT_DIR="${HOME}/.git"
```

Comment:
Not sure about the reason behind this change. We set `HOME` in order to safely run `git set --global`. The `.git` directory should normally be under `PYENV_INSTALL_PATH` (but we probably cannot fake it lest scripts won't be able to find stuff). AFAICS if `GIT_DIR` is set, `git` uses it regardless of where it is -- so just setting it and creating the fake directory as necessary should successfully mock the git-related logic.